### DISCUSSION TITLE
Checkboxes of File associations list are confusing

### DISCUSSION THREAD

**Discussion by mkahvi** [d2phap/ImageGlass#343] [General] [answered]
### System info:
- ImageGlass version: 5.0.5.7
- OS version: Windows 7 64-bit

### Expected behavior:
File associations are saved.

### Actual behavior:
File associations are not saved.
Settings window simply closes after saving and on subsequent opening or restarting of IG the associations are empty again.

### Steps to reproduce:
1. Start IG, Go to settings -> File associations
2. Select anything
3. Apply & save

### Other info:
This happens with both portable and non-portable, though I assume it's expected for the former.
Running as admin makes no difference (except that my other settings are different for some reason?).

**Comment by d2phap** [comment]
Hi @mkahvi 

I think it's a bit confusing that the checkboxes are just for "selection mark" only, means when you select a row, the corresponding checkbox is also selected.

I will remove the checkboxes in next version.

All the file extensions that showing in the list are "saved" already. You can add and remove it.

**Comment by d2phap** [comment]
removed checkboxes

**Comment by mkahvi** [comment]
Even if I have them selected only, without touching the checkboxes, the associations are not saved. Everything still keeps being opened by the previously associated image viewer, which does not automatically re-associate them.

**Comment by d2phap** [comment, marked as answer]
If you make change of the extensions list, you will need to click on _Set as Default photo viewer_ button to register to Windows.

Then ImageGlass will appear in the list of default app.


Windows 7, 8:
![image](https://user-images.githubusercontent.com/3154213/39976786-1a9cc412-5768-11e8-9435-4720c531bdb3.png)




Windows 10:
![image](https://user-images.githubusercontent.com/3154213/39976533-d2ab55a2-5766-11e8-8416-330e3770cc6a.png)
